File: Data Structures & Algorithms/course-schedule-iv/submission-1.py
class Solution:
    def checkIfPrerequisite(self, numCourses: int, prerequisites: List[List[int]], queries: List[List[int]]) -> List[bool]:
        preMap={i:[] for i in range(numCourses)}
        ans=[]
        li=[]
        for crs,pre in prerequisites:
            preMap[crs].append(pre)
        
        prereqMap={}
        def dfs(crs):
            if crs not in prereqMap:
                prereqMap[crs]=set()
                for pre in preMap[crs]:
                    prereqMap[crs]|=dfs(pre)
                prereqMap[crs].add(crs)
            return prereqMap[crs]
        
        for i in range(numCourses):     
            dfs(i)

        for u,v in queries:
            ans.append(v in prereqMap[u])
        
        return ans

File: Data Structures & Algorithms/concatenation-of-array/submission-1.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n=nums.size();
        vector<int> ans(2*n);
        
        int i=0;
        while(i<n){
            ans[i]=nums[i];
            i++;
        }

        i=0;
        while((i+n)<(2*n)){
         ans[i+n]=nums[i];
         i++;
        } 
        
        return ans;
        }
};